libraries/AP_L1_Control/AP_L1_Control.cpp

```diff
@@ -141,10 +141,8 @@ float AP_L1_Control::loiter_radius(const float radius) const
     float sanitized_bank_limit = constrain_float(_loiter_bank_limit, 0.0f, 89.0f);
     float lateral_accel_sea_level = tanf(radians(sanitized_bank_limit)) * GRAVITY_MSS;
 
-    float nominal_velocity_sea_level;
-    if(_tecs == nullptr) {
-        nominal_velocity_sea_level = 0.0f;
-    } else {
+    float nominal_velocity_sea_level = 0.0f;
+    if(_tecs) {
```

Comment:
```suggestion
    if (_tecs != nullptr) {
```

Reply:
In the other class, the pointer is determined as follows.
If you want to make it explicit that it is a NULLPTR, there are several processes as follows.
I would like to refactor

if (optflow && optflow->enabled()) {

Reply:
> In the other class, the pointer is determined as follows. If you want to make it explicit that it is a NULLPTR, there are 

We're not really stick about it, but we shouldn't remove the nullptr stuff if it is already around in the code (which you patch here is doing.

This is more an "improve code over time" sort of thing rather than a go-out-and-find-places-that-could sort of a thing.

> several processes as follows. I would like to refactor
> 
> if (optflow && optflow->enabled()) {

We don't really need to chase these down to fix them; time is better spent on doing things like making more things build-time options; we've got real problems on some of our boards at the moment, so more build-time options could be immediately useful.

(also note the addition of the space after the `if`.  I might not even have commented if there hadn't been two problems on the one line :-) )

Reply:
I have changed the comparison to be done in NULLPTR.